compiler/rustc_expand/src/base.rs

```diff
@@ -1032,6 +1032,15 @@ impl<'a> ExtCtxt<'a> {
         self.sess.parse_sess.span_diagnostic.struct_span_err(sp, msg)
     }
 
+    pub fn struct_span_err_with_code<S: Into<MultiSpan>>(
+        &self,
+        sp: S,
+        msg: &str,
+        code: DiagnosticId,
+    ) -> DiagnosticBuilder<'a> {
+        self.sess.parse_sess.span_diagnostic.struct_span_err_with_code(sp, msg, code)
+    }
```

Comment:
Is this necessary?

Reply:
I think it is good for feature parity with `rustc_errors`.

Reply:
The only thing I don't ilke it it adds yet another `struct_span_err_with_code` to the docs, which make the original version harder to find.

Reply:
I don't see why it needs to be added to `rustc_expand` directly though. You could just use `struct_span_err!` which makes this nice and easy to type: `struct_span_err!(self.sess, sp, msg, E0774)`.

Reply:
Maybe we should remove `struct_span_err` above later?

Reply:
I am fine with either but if we follow `struct_span_err!` can we remove `struct_span_err` that was already there in favor of `struct_span_err!` later on?

Reply:
I'm opening an issue for that. I think it's better to do it in another PR.